### PR TITLE
Ticket7

### PR DESCRIPTION
**Pull request by AdilBulsari**

### CODE REVIEW

File: src/Components/Topics/Vote.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { patchVoteById } from '../API/Api'
+import Button from './Button'
+import styles from './Vote.module.css'
+
+function Vote(props) {
+    const [disableLikeButton, setDisableLikeButton] = useState(false)
+    const [disableUnLikeButton, setDisableUnLikeButton] = useState(true)
+    const [error, setError] = useState(false)
+
+    function increaseVoteHandlerApi() {
```
Comment (Angaca) [inline src/Components/Topics/Vote.jsx:11]: This can be optimised in a single function

File: src/Components/Topics/Vote.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        setDisableLikeButton((prevState) => !prevState)
+    }
+
+    function decreaseVoteHandlerApi() {
+        setDisableLikeButton((prevState) => !prevState)
+        props.setVotes((prevVote) => prevVote - 1)
+        patchVoteById(props.article.article_id, -1).catch((err) => {
+            setError(true)
+        })
+        setDisableUnLikeButton((prevState) => !prevState)
```
Comment (Angaca) [inline src/Components/Topics/Vote.jsx]: This can be optimised in a single function

**Review by Angaca — APPROVED** [review]
If you are happy with the user not being able to downvote but only eventually remove the like this is fine, really your choice. I am still unable to see a single article card and vote if I see all the articles page

File: src/Components/Article/ArticleDescription.jsx [hunk 3]
```diff
@@ -37,11 +40,17 @@ function ArticleDescription() {
...
                     </p>
                     <p>
                         <b>Comments made :</b> {article.total_comments}
                     </p>
+                    <Vote
+                        article={article}
+                        votes={votes}
+                        setVotes={setVotes}
+                        setArticle={setArticle}
+                    />
```
Comment (douglashellowell) [inline src/Components/Article/ArticleDescription.jsx]: the only prop you're actually using here is `setVotes` - remove these props when you stop using them

**Comment by douglashellowell** [comment]
Nice work!

---

I like the cursor changing when the button is disabled but users don't have cursors on mobile devices! You should indicate that the button is disabled visually using some CSS as well

---

Your components become too narrow to view on smaller screens - check your CSS!

image

---

Your React app is a HTML website - not an api! You don't need to add `/api` on your paths

You don't need to name your FE endpoints by a function name either - they can just be what resource they are

```diff
- /api/getAllTopics
+ /topics

- /api/getTopicArticle/:topic_name
+ /articles/:topic_name
```
> etc etc

File: src/Components/Topics/Vote.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            </p>
+            <Button
+                disabled={disableLikeButton ? 'disabled' : ''}
+                onClick={() => VoteHandlerApi(+1)}
+            >
+                <img
+                    className={styles.like}
+                    alt="like"
+                    src={require('../Tools/LikeSVG/likeicon.png')}
+                />
```
Comment (douglashellowell) [inline src/Components/Topics/Vote.jsx]: you're mixing import styles in this file which is a bit unusual

You should only be using `import * from './path'` module syntax 

you can import this image at the top with the rest of your imports and then put it in `src={}`

File: src/Components/Topics/Vote.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        setDisableUnLikeButton((prevState) => !prevState)
+    }
+
+    return (
+        <div className={styles.vote}>
+            <p>
+                <b>CLICK TO VOTE</b>
+            </p>
+            <Button
+                disabled={disableLikeButton ? 'disabled' : ''}
```
Comment (douglashellowell) [inline src/Components/Topics/Vote.jsx]: > This **Boolean** attribute prevents the user from interacting with the button: it cannot be pressed or focused.
(MDN)

You're passing a string!

`disableLikeButton` is already a boolean so you should be able to just pass in that

File: src/Components/Topics/Vote.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react'
+import { patchVoteById } from '../API/Api'
+import Button from './Button'
+import styles from './Vote.module.css'
+
+function Vote(props) {
+    const [disableLikeButton, setDisableLikeButton] = useState(false)
+    const [disableUnLikeButton, setDisableUnLikeButton] = useState(true)
```
Comment (douglashellowell) [inline src/Components/Topics/Vote.jsx:10]: the downvote button is disabled at the start!

**Review by Angaca — APPROVED** [review]
You have a small bug when I try to dislike at first it says I liked the post, a part of that is mainly working even if I can only like or dislike I cannot completely change my mind and like after I initially disliked or the opposite. Design choice and you can leave like this if this is what you imagined

**Pull request merged**